ccswbs/doorknob
# First prototype of admission requirements filter

# Summary of changes
A first prototype of the frontend for the admission requirements filter. Right now only very basic data exists to test the functionality as I believe we should wait until the backend is complete before we worry about entering all the data (because there is a lot :) ). Note that some of the code will change when integrating the backend.

Also I dont think this code should be merged until the backend is complete. This PR is more so just to showcase what my approach to the problem was and whether it seems as though it is a viable solution or if there is room to improve.

Check the code to see how the filter functions and how pages are built. If something isn't clear, don't hesitate to ask me for some clarification

## Frontend
- Added requirements page template (src/templates/admission/requirements.js)
- Added requirements filter form page (src/pages/admission/requirements/index.js)
- Added custom resolver and createPages functionality to gatsby-node.js (Note this will change once we have completed the backend).
- Added several yaml files representing the data for the requirements under src/data/admission (subject to change once the backend is complete)
- Added util functions in various files under src/utils (you can ignore toTitleCase.js and slugify.js for now)

[x] My changes are accessible (at minimum WCAG 2.0 Level AA)
[x] My changes are responsive and appear as expected on mobile and desktop views.

## Backend
- N/A

# Test Plan

Preview link: https://deploy-preview-50--doorknob.netlify.app/admission/requirements

1. Go to the preview link.
 
2. When an option is selected for the "I'm interested in" dropdown, ensure the options that show up in the following "Choose your desired field" dropdown make sense. (i.e. if Bachelor's Degree is selected then the "Choose your desired field" dropdown only shows bachelor degrees.

3. Select the following in the dropdowns: 
 
  Ontario
  High School Student
  Bachelor's Degree
  Bachelor of Engineering

  and view the page: there should be a general requirements section and a program specific section.

4. Select the following in the dropdowns: 
 
  Ontario
  High School Student
  Bachelor's Degree
  Bachelor of Computing

  and view the page: there should only be a general requirements which will be identical to the one on the Bachelor of Engineering page and the URL should be /admission/requirements/undergraduate/high-school/ontario/. 

5. Play around with other options, note most pages will be blank as the data isn't there, just make sure none of the options leads to a 404.
